service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java

```diff
@@ -0,0 +1,228 @@
+    }
+    certificates.forEach(c -> c.setTags(Collections.emptyList()));
+    List<CertificateDto> certificateDtoList =
+        certificates.stream().map(certificateInterpreter::toDto).collect(Collectors.toList());
+    certificateDtoList.forEach(o -> o.setTags(tagService.findByCertificateId(o.getId())));
+    return certificateDtoList;
+  }
+
+  @Override
+  @Transactional
+  public CertificateDto findById(String id) {
+    List<String> errors = StringIdValidator.idValidate(id);
+    if (!errors.isEmpty()) {
+      throw new ValidationException(errors, CustomHttpStatus.BAD_REQUEST.CERTIFICATE);
+    }
+    long valueId = Long.parseLong(id);
+    Certificate certificate =
+        certificateDao
+            .findById(valueId)
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
+                        List.of(NOT_FOUND_CERTIFICATE.formatted(valueId)),
+                        CustomHttpStatus.NOT_FOUND.CERTIFICATE));
+    CertificateDto certificateDto = certificateInterpreter.toDto(certificate);
+    certificateDto.setTags(tagService.findByCertificateId(valueId));
+    return certificateDto;
+  }
+
+  @Override
+  @Transactional
+  public CertificateDto create(CertificateDto certificateDto) {
+    List<String> errors = CertificateDtoValidator.createCertificateDtoValidate(certificateDto);
+    if (!errors.isEmpty()) {
+      throw new ValidationException(errors, CustomHttpStatus.BAD_REQUEST.CERTIFICATE);
+    }
+    errors = tagsIsExists(certificateDto.getTags());
+    if (!errors.isEmpty()) {
+      throw new NotFoundException(errors, CustomHttpStatus.NOT_FOUND.TAG);
+    }
+    Certificate certificate =
+        certificateDao
+            .save(certificateInterpreter.fromDto(certificateDto))
+            .orElseThrow(
+                () ->
+                    new CreationException(
+                        NOT_CREATED_CERTIFICATE, CustomHttpStatus.CONFLICT.CERTIFICATE));
+    certificate.setTags(Collections.emptyList());
+    certificateDto.setTags(new HashSet<>(certificateDto.getTags()).stream().toList());
+    CertificateDto certificateDtoResult = certificateInterpreter.toDto(certificate);
+    certificateDtoResult.setTags(tagService.createTagsOnlyByNameOrId(certificateDto.getTags()));
+    certificateDtoResult
+        .getTags()
+        .forEach(
+            tagDto ->
+                certificateDao.addTagToCertificate(certificateDtoResult.getId(), tagDto.getId()));
+    return certificateDtoResult;
+  }
+
+  @Override
+  @Transactional
+  public void removeById(String id) {
+    List<String> errors = StringIdValidator.idValidate(id);
+    if (!errors.isEmpty()) {
+      throw new ValidationException(errors, CustomHttpStatus.BAD_REQUEST.CERTIFICATE);
+    }
+    certificateDao.removeById(Long.parseLong(id));
+  }
+
+  @Override
+  @Transactional
+  public CertificateDto update(String id, CertificateDto certificateDto) {
```

Comment:
You call this method in the patch update method.
Here you use 
 _certificateDao
            .update(certificateInterpreter.fromDto(certificateDto))_
which, as far as I can see read all fields(even not specified in the request), and pass them in dao-update method.

So, is this really a patch update then?

Reply:
Although the whole object is passed to the method, only those fields that are initialized are updated in the database.